reactor-core/src/main/java/reactor/core/publisher/FluxScanSeed.java

```diff
@@ -147,90 +125,74 @@ public void onError(Throwable t) {
 		}
 
 		@Override
-		public void onComplete() {
+		public void onNext(T t) {
 			if (done) {
+				Operators.onNextDropped(t);
 				return;
 			}
-			done = true;
 
-			R v = value;
+			R r = value;
 
-			long p = produced;
-			long r = requested;
-			if (r != Long.MAX_VALUE && p != 0L) {
-				r = REQUESTED.addAndGet(this, -p);
+			try {
+				r = Objects.requireNonNull(accumulator.apply(r, t),
+						"The accumulator returned a null value");
+			}
+			catch (Throwable e) {
+				onError(Operators.onOperatorError(s, e, t));
+				return;
 			}
 
-			for (; ; ) {
-				// if any request amount is still available, emit the value and complete
-				if ((r & REQUESTED_MASK) != 0L) {
-					actual.onNext(v);
-					actual.onComplete();
-					return;
-				}
-				// NO_REQUEST_NO_VALUE -> NO_REQUEST_HAS_VALUE
-				if (REQUESTED.compareAndSet(this, 0, COMPLETED_MASK)) {
-					return;
-				}
+			actual.onNext(r);
+			value = r;
+		}
+
+		@Override
+		public void onSubscribe(Subscription s) {
+			if (Operators.validate(this.s, s)) {
+				this.s = s;
 
-				r = requested;
+				actual.onSubscribe(this);
 			}
 		}
 
 		@Override
 		public void request(long n) {
-			if (Operators.validate(n)) {
-				for (; ; ) {
-
-					long r = requested;
-
-					// NO_REQUEST_HAS_VALUE 
-					if (r == COMPLETED_MASK) {
-						// any positive request value will do here
-						// transition to HAS_REQUEST_HAS_VALUE
-						if (REQUESTED.compareAndSet(this,
-								COMPLETED_MASK,
-								COMPLETED_MASK | 1)) {
-							actual.onNext(value);
-							actual.onComplete();
-						}
-						return;
-					}
-
-					// HAS_REQUEST_HAS_VALUE
-					if (r < 0L) {
-						return;
-					}
-
-					// transition to HAS_REQUEST_NO_VALUE
-					long u = Operators.addCap(r, n);
-					if (REQUESTED.compareAndSet(this, r, u)) {
-						s.request(n);
-						return;
-					}
+			if (STATE.compareAndSet(this, 0, SENT_SEED)) {
```

Comment:
This won't work properly with two concurrent requests because both of them may trigger a concurrent call to `onNext`. The safe solution is to use queue-drain where the queue contains a the initial value to begin with: [FlowableScanSeed](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/internal/operators/flowable/FlowableScanSeed.java)